sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -459,7 +462,7 @@ class SQLContext(@transient val sparkContext: SparkContext)
       iter.map { row =>
         new GenericRow(
           extractors.zip(attributeSeq).map { case (e, attr) =>
-            DataTypeConversions.convertJavaToCatalyst(e.invoke(row), attr.dataType)
+            CatalystTypeConverters.convertToCatalyst(e.invoke(row), attr.dataType)
```

Comment:
@marmbrus, should we consider this case (`def createDataFrame(rdd: RDD[_], beanClass: Class[_]): DataFrame`) sufficiently rare to not optimize?